src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -46,6 +53,10 @@ public void addNode(T node) {
     m_nodesReady.add(node);
   }
 
+  public void addEdge(int weight, T from, T to, int order) {
```

Comment:
+1 but I put `weight` as the first param because I wanted to keep a coherence with `public void addEdge(int weight, T from, T... tos)` where `tos` must be at the end.

I don't know what could be the best solution.